xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm

```diff
@@ -768,16 +768,15 @@ $fn
 #macro(panelheader $title)
 ## Note: We pass the Panel name as an HTML class attribute so that it's possible to style
 ## the Panels selectively using CSS.
-#set ($specialClassAttribute = '')
-#if ($paneldoc)
-  #set ($specialClassAttribute = $util.convertToAlphaNumeric($paneldoc.name))
-#end
+#set ($specialClassAttribute = $stringtool.toAlphaNumeric("$!paneldoc.documentReference"))
```

Comment:
For the serialization you should use instead: `$!services.model.serialize($paneldoc.documentReference)`

Reply:
Actually we should not serialize the wiki name IMO. So we should use a `compactwiki` serializer, i.e. `$!services.model.serialize($paneldoc.documentReference, 'compactWiki')`

Reply:
+1 to use the last space **name** if the panel page is non-terminal, and the page **name** if the panel page is terminal. I know that technically we can have panels with the same (page) name at different locations but this is an edge case and I prefer not to have the panel location (full path) in the class name so that moving the panel doesn't require to update the CSS.

+0.5 to serialize (using compact wiki) the last space **reference** if the panel page is non-terminal, and the page **reference** if the panel page is terminal.